### PR TITLE
Fix #472: Fail identity verification after failed OTP

### LINKED ISSUE #472: Fail identity verification after failed OTP
When SCA OTP fails, status endpoint returns `phase=null, status=NOT_INITIALIZED` (this is correct) but identity verification stays in `phase=OTP_VERIFICATION, status=VERIFICATION_PENDING` in the database.

### PR DESCRIPTION
**Pull request by banterCZ**

### CODE REVIEW

File: enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/database/IdentityVerificationRepository.java [hunk 1]
```diff
@@ -39,17 +39,6 @@
 @Repository
 public interface IdentityVerificationRepository extends CrudRepository<IdentityVerificationEntity, String> {
 
-    @Modifying
```
Comment (banterCZ) [inline enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/database/IdentityVerificationRepository.java:42]: I believe that it is not needed anymore, when reset sets status to FAILED.

File: enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/service/IdentityVerificationLimitService.java [hunk 2]
```diff
@@ -124,45 +123,51 @@ public void checkIdentityVerificationLimit(OwnerId ownerId) throws RemoteCommuni
...
-            identityVerification.setStatus(IdentityVerificationStatus.FAILED);
-            identityVerification.setErrorDetail(IdentityVerificationEntity.ERROR_MAX_FAILED_ATTEMPTS_DOCUMENT_UPLOAD);
-            identityVerification.setErrorOrigin(ErrorOrigin.PROCESS_LIMIT_CHECK);
-            identityVerification.setTimestampLastUpdated(ownerId.getTimestamp());
-            identityVerification.setTimestampFailed(ownerId.getTimestamp());
-            identityVerificationRepository.save(identityVerification);
-            logger.info("Switched to {}/FAILED; {}", phase, ownerId);
-            auditService.audit(identityVerification, "Switched to {}/FAILED; user ID: {}", phase, ownerId.getUserId());
-            resetIdentityVerification(ownerId);
+            resetIdentityVerification(ownerId, ErrorOrigin.PROCESS_LIMIT_CHECK, IdentityVerificationEntity.ERROR_MAX_FAILED_ATTEMPTS_DOCUMENT_UPLOAD);
```
Comment (banterCZ) [inline enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/service/IdentityVerificationLimitService.java:141]: Not sure whether `ErrorOrigin.PROCESS_LIMIT_CHECK` is correct for `documentUploadMaxFailedAttempts`, but it was already there.

File: enrollment-server-onboarding-domain-model/src/main/java/com/wultra/app/enrollmentserver/model/enumeration/ErrorOrigin.java [hunk 3]
```diff
@@ -32,9 +32,13 @@ public enum ErrorOrigin {
 
     OTP_VERIFICATION,
 
+    USER_VERIFICATION,
```
Comment (banterCZ) [inline enrollment-server-onboarding-domain-model/src/main/java/com/wultra/app/enrollmentserver/model/enumeration/ErrorOrigin.java:35]: When these two new error origins approved, I will change the documentation.

File: enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/database/IdentityVerificationRepository.java (hunk 1 above)
Comment (romanstrobl) [reply enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/database/IdentityVerificationRepository.java:42]: Ok.

File: enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/service/IdentityVerificationLimitService.java (hunk 2 above)
Comment (romanstrobl) [reply enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/service/IdentityVerificationLimitService.java:141]: Yes, it is fine, all limit checks are under this one.

File: enrollment-server-onboarding-domain-model/src/main/java/com/wultra/app/enrollmentserver/model/enumeration/ErrorOrigin.java (hunk 3 above)
Comment (romanstrobl) [reply enrollment-server-onboarding-domain-model/src/main/java/com/wultra/app/enrollmentserver/model/enumeration/ErrorOrigin.java:35]: Ok.

**Review by romanstrobl — APPROVED** [review]
Looks OK.

**Pull request merged**